### ISSUE TITLE
Save checkpoint before learning rate drop

### ISSUE THREAD

**Issue by fredzzhang** [fredzzhang/pocket#37]
Save checkpoint before calling `scheduler.step()` to ensure the saved optimiser state dict contains the learning rate used in the elapsed epoch.
